azure_li_services/units/system_setup.py

```diff
@@ -180,6 +190,81 @@ def set_kdump_service(config, status):
     )
 
 
+def set_stonith_service(config):
+    # 1. setup InitiatorName
+    initiator_name = 'iqn.1996-04.de.suse:01:{0}'.format(
+        config['initiatorname']
+    )
+    initiator_file = '/etc/iscsi/initiatorname.iscsi'
+    initiator_data = re.sub(
+        r'InitiatorName=.*', 'InitiatorName={0}'.format(initiator_name),
+        _read_file(initiator_file)
+    )
+    _write_file(initiator_file, initiator_data)
+
+    # 2. setup iscsi options
+    iscsi_conf = '/etc/iscsi/iscsid.conf'
+    iscsi_conf_data = _read_file(iscsi_conf)
+    iscsi_conf_data = re.sub(
+        r'node.session.timeo.replacement_timeout = .*',
+        'node.session.timeo.replacement_timeout = 5',
+        iscsi_conf_data
+    )
+    iscsi_conf_data = re.sub(
+        r'node.startup = .*', 'node.startup = automatic',
+        iscsi_conf_data
+    )
+    _write_file(iscsi_conf, iscsi_conf_data)
+
+    # 3. initialize iSCSI devices
+    discovery_output = Command.run(
+        [
+            'iscsiadm', '-m', 'discovery',
+            '-t', 'st', '-p', '{0}:3260'.format(config['ip'])
+        ]
+    ).output.splitlines()
+
+    # 4. create sbd device
+    # The first line of the iscsiadm discovery output is taken into
+    # account as we expect any group tag to respond with the same
+    # target name. We also expect the discovery output to be in the
+    # format: target_IP:port,target_portal_group_tag proper_target_name
+    discovery_format = '.*:.*,.* .*'
+    if discovery_output and re.match(discovery_format, discovery_output[0]):
+        target_name = discovery_output[0].split(' ')[1]
+        target_device = '/dev/disk/by-path/ip-{0}:3260-iscsi-{1}-lun-0'.format(
+            config['ip'], target_name
+        )
+        Command.run(
+            ['iscsiadm', '-m', 'node', '-l']
+        )
+        Command.run(
+            ['rescan-scsi-bus.sh']
+        )
+        Command.run(
+            ['sbd', '-d', target_device, 'create']
+        )
+        Command.run(
+            ['sbd', '-d', target_device, 'dump']
```

Comment:
No, in case of an error the stderr information will be part of the exception data which ends up in the systemd log. Is the output of the commands somehow relevant ?

Reply:
ok, I took the time reading the complete setup procedure for the sbd device. It seems device information is needed for other steps later. If we should write details about the ```sbd``` calls please let me know what exactly is needed and how you expect the information. Thanks

Reply:
@rkrishna12 any thoughts ?

Reply:
@schaefi, Can you add this step as well? Then, the customer will be able to easily identify the SBD device and continue with the rest of the procedure.

> Update /etc/sysconfig/sbd with this line:
>  SBD_DEVICE="\<SBD Device Name\>"

 systemd logs for any errors will be sufficient.

Reply:
yep will do, makes much sense to me. Thanks

Reply:
@rkrishna12 done now, please double check. Thanks